packages/server/src/api/routes/tests/automation.spec.ts

```diff
@@ -477,4 +478,115 @@ describe("/automations", () => {
       }
     )
   })
+  describe("Automation Update / Creator row trigger filtering", () => {
+    let table: Table
+
+    beforeAll(async () => {
+      table = await config.createTable({
+        name: "table",
+        type: "table",
+        schema: {
+          Approved: {
+            name: "Approved",
+            type: FieldType.BOOLEAN,
+          },
+        },
+      })
+    })
+
+    const testCases = [
+      {
+        description: "should run when Approved changes from false to true",
+        filters: {
+          equal: { "1:Approved": true },
+        },
+        row: { Approved: "true" },
+        oldRow: { Approved: "false" },
+        onlyExecuteFilterIfChange: true,
+        expectToRun: true,
+      },
+      {
+        description:
+          "should not run when Approved is true in both old and new row",
+        filters: {
+          equal: { "1:Approved": true },
+        },
+        row: { Approved: "true" },
+        oldRow: { Approved: "true" },
+        onlyExecuteFilterIfChange: true,
+        expectToRun: false,
+      },
+      {
+        description:
+          "should run when Approved is true in both old and new row (onlyExecuteFilterIfChange: false)",
+        filters: { equal: { "1:Approved": true } },
+        row: { Approved: "true" },
+        oldRow: { Approved: "true" },
+        onlyExecuteFilterIfChange: false,
+        expectToRun: true,
+      },
+
+      {
+        description:
+          "should run when a contains filter matches the correct options",
+        filters: {
+          contains: { "1:opts": ["Option 1", "Option 3"] },
+        },
+        row: { opts: ["Option 1", "Option 3"] },
+        oldRow: { opts: ["Option 3"] },
+        onlyExecuteFilterIfChange: true,
+        expectToRun: true,
+      },
+      {
+        description:
+          "should not run when opts doesn't contain any specified option",
+        filters: {
+          contains: { "1:opts": ["Option 1", "Option 2"] },
+        },
+        row: { opts: ["Option 3", "Option 4"] },
+        oldRow: { opts: ["Option 3", "Option 4"] },
+        onlyExecuteFilterIfChange: true,
+        expectToRun: false,
+      },
+    ]
+
+    it.each(testCases)(
+      "$description",
+      async ({
+        filters,
+        row,
+        oldRow,
+        onlyExecuteFilterIfChange,
+        expectToRun,
+      }) => {
+        let automation = await updateRowAutomationWithFilters()
```

Comment:
Don't think you need this await. Also, given that this structure is only used in this one location, why not define it inline? It seems strange to me that this definition lives quite far away from its single use in the codebase.